DomainContent/AvatarStore/CheckoutZone/checkoutZone.js

```diff
@@ -0,0 +1,255 @@
+                spawnX = 0;
+                return;
+            }
+        });
+    });
+
+    // Spawn a copy of each attachment scaled down to fit the ITEM_HEIGHT and place it on the checkout table
+    var spawnOverlayReplica = (function(entityID) {
+        var entityProperties = Entities.getEntityProperties(entityID, [
+            'name', 'modelURL', 'type', 'dimensions', 'marketplaceID', 'modelURL',
+            'localPosition', 'localRotation', 'dimensions', 'parentJointIndex'
+        ]);
+        var overlayProperties = {
+            url: entityProperties.modelURL,
+            name: OVERLAY_PREFIX + entityProperties.marketplaceID,
+            alpha: true,
+            grabbable: true,
+            parentID: tableID,
+            localPosition: {x: spawnX, y: spawnY, z: spawnZ},
+            localRotation: {x: 0, y: 0, z: 0},
+            // clone dimensions so we can alter it without messing up the original entities dimensions
+            dimensions: JSON.parse(JSON.stringify(entityProperties.dimensions))
+        };
+        var scale = (ITEM_HEIGHT / overlayProperties.dimensions.y);
+        if ((overlayProperties.dimensions.x > ITEM_HEIGHT) || (overlayProperties.dimensions.y > ITEM_HEIGHT) || (overlayProperties.dimensions.y > ITEM_HEIGHT)) {
+            overlayProperties.dimensions.y = ITEM_HEIGHT;
+            overlayProperties.dimensions.x *= scale;
+            overlayProperties.dimensions.z *= scale;
+        }
+        // check that the item is not too large
+        var maxItemSize = 0.3;
+        var scaleReduction;
+        while (overlayProperties.dimensions.x > maxItemSize || overlayProperties.dimensions.z > maxItemSize) {
+            scale *= scaleReduction;
+            overlayProperties.dimensions.y *= scale;
+            overlayProperties.dimensions.x *= scale;
+            overlayProperties.dimensions.z *= scale;
+        }
+        var replica = Overlays.addOverlay("model", overlayProperties);
+        var replicaStoredTransform = {
+            position: entityProperties.localPosition,
+            rotation: entityProperties.localRotation,
+            dimensions: entityProperties.dimensions,
+            jointName: MyAvatar.jointNames[entityProperties.parentJointIndex],
+            demoEntityID: entityID
+        };
+
+        print('stored transform = ' + JSON.stringify(replicaStoredTransform));
+
+        replicaStoredTransforms[replica] = replicaStoredTransform;
+        replicaList.push(replica);
+    });
+
+    _this.replicaCheckedOut = function(entityID, args) {
+        var ARGS_INDEX = {
+            REPLICA_OVERLAY: 0,
+            NEW_ENTITY: 1
+        };
+        var replicaOverlayID = args[ARGS_INDEX.REPLICA_OVERLAY];
+        var newEntityID = args[ARGS_INDEX.NEW_ENTITY];
+        print('replicaOverlayID = ' + replicaOverlayID);
+        print('newEntityID = ' + newEntityID);
+        
+        // Delete the new entity when the transforms are not found.
```

Comment:
We should discuss this behavior -- if something is incorrectly setup with the transform, this can be hard to troubleshoot and occasionally can cause an unexpected user flow